### PR TITLE
Fix bug layout

### PR DESCRIPTION
**Pull request by akaka0039**
## Related Issue
fixed

### CODE REVIEW

File: src/app/Http/Requests/StoreEventRequest.php [hunk 1]
```diff
@@ -21,16 +22,24 @@ public function authorize(): bool
...
+            'start_date' => ['required', 'date_format:Y-m-d H:i'],
+            'end_date' => ['required', 'date_format:Y-m-d H:i', 'after:start_date'],
+            'is_online' => ['bool', 'required'],
         ];
+
+        if ($this->input('is_online')) {
+            // true_online
+            $rules['address'] = ['required', 'url'];
+        } else {
+             // true_inPerson
```
Comment (akaka0039) [inline src/app/Http/Requests/StoreEventRequest.php:40]: should change this comment

File: src/app/Http/Requests/StoreEventRequest.php [hunk 2]
```diff
@@ -21,16 +22,24 @@ public function authorize(): bool
      */
     public function rules(): array
     {
-        return [
+        $rules = [
             'user_id' => ['required', 'integer'],
             'name' => ['required', 'string', 'max:255'],
-            'address' => ['nullable', 'string', 'max:255'],
+            'address' => ['required', 'string', 'max:255'],
```
Comment (shin5446) [inline src/app/Http/Requests/StoreEventRequest.php:28]: You might want to write it this way.
```
 'address' => $this->input('is_online') ? ['required', 'url'] : ['required', 'string', 'max:255'],
```

Comment (akaka0039) [reply src/app/Http/Requests/StoreEventRequest.php:28]: Thank you for the comment!!!

File: src/app/Http/Controllers/EventController.php [hunk 3]
```diff
@@ -25,18 +26,48 @@ public function index()
...
+        // if image is there
+        if ($request->hasFile('image')) {
+            $original = $request->file('image')->getClientOriginalName();
+            $name = date('Ymd_His') . '_' . $original;
+            $request->file('image')->move('storage/images', $name);
+            $event->image = $name;
+        } else {
+            // default
+            $event->image = 'top_background.jpg'; 
+        }
```
[the remark this answers is not in the document]
Comment (shin5446) [reply src/app/Http/Controllers/EventController.php:57]: And then, you can write like this in the store action in EventController.
```
    public function store(StoreEventRequest $request)
    {
        $validated = $request->validated();
        $validated['image'] = $request->get('image');

        Event::create($validated);
        return redirect('/events')->with('message', 'Your event has been successfully created!');
    }
```
The validated function does not give the value of the formatted image, so I add the formatted image in the controller.

[the remark this answers is not in the document]
Comment (akaka0039) [reply src/app/Http/Controllers/EventController.php:57]: Thank you so much!!!

I tried this and I have one question now.
I found that 'image' columns is null if a user didn't upload a image of a event.

do you know why?

[the remark this answers is not in the document]
Comment (shin5446) [reply src/app/Http/Controllers/EventController.php:57]: And then, you can write like this in the store action in EventController.
```
    public function store(StoreEventRequest $request)
    {
        $validated = $request->validated();
        $validated['image'] = $request->get('image');

        Event::create($validated);
        return redirect('/events')->with('message', 'Your event has been successfully created!');
    }
```
The validated function does not give the value of the formatted image, so I add the formatted image in the controller l

[the remark this answers is not in the document]
Comment (akaka0039) [reply src/app/Http/Controllers/EventController.php:57]: I mean the both I wrote down 
does the code correctly work in your local?

**Pull request merged**